State Pivot Tables

jwal1024 opened the pull request:
Improvements to function for reading data from GitHub, use of new microdf groupby to simplify functions, pivot tables of state-based % changes in poverty and gini.

jwal1024 committed:
State Pivot Tables

Improvements to function for reading data from GitHub, use of new microdf groupby to simplify functions, pivot tables of state-based % changes in poverty and gini.

jwal1024 committed:
Merge pull request #4 from jwal1024/State_analysis

State Pivot Tables

jwal1024 committed:
Delete child_allowance_costs.py

pep8speaks commented:
Hello @jwal1024! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`jb/child_allowance.py`](https://github.com/UBICenter/child-allowance/blob/f243004c9fb42992a1b08bc15d80ce226de2a413/jb/child_allowance.py):

> [Line 148:1](https://github.com/UBICenter/child-allowance/blob/f243004c9fb42992a1b08bc15d80ce226de2a413/jb/child_allowance.py#L148): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -1,17 +1,13 @@
-raw = pd.read_csv(
-    "https://github.com/UBICenter/child-allowance/blob/master/jb/data/cps_00003.csv.gz",
+person_raw = pd.read_csv(
+    "https://github.com/UBICenter/child-allowance/blob/master/jb/data/cps_00003.csv.gz?raw=true",  # noqa
```
Load this locally (should also be able to skip `compression="gzip"` after doing so

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -150,72 +145,116 @@
+    )
+
+
+# Gini coefficients and state/demographic-based heterogenous gini coefficients
```
I haven't really seen demographic-based Gini coefficients (I don't think people care too much about inequality within the Black population), I'd say this is low priority

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -150,72 +145,116 @@
+
+
+# Define percentage change functions
+def pp_change(base, new):
```
I'd skip this being a function and just subtract when it's needed, otherwise readers will have to look at the function to see if it's something more complex than it appears to need

MaxGhenis reviewed (commented):
Some minor comments. Also it'd be good for this to do something that feeds into the broader analysis by exporting results for the notebook to load and display. It currently looks like it prints a couple dataframes but this would only be useful when viewing from terminal or copy/pasting elsewhere.

MaxGhenis commented on jb/child_allowance.py:
```diff
@@ -150,72 +145,116 @@
+state_gini["gini_change_flat"] = pp_change(
+    state.baseline, state.child_allowance
+)
+state_gini["gini_change_%_cc"] = percent_change(
```
avoid special characters like `%` in column names

MaxGhenis reviewed (commented):
.

jwal1024 committed:
Resolve Max Comments

Removed demographic Ginis but retained state-based; removed pp_change function; removed special characters.

Remaining: Deal with local load + compression
